src/test/java/echoserver/ClientServerConnectionTest.java

```diff
@@ -0,0 +1,51 @@
+package echoserver;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.net.Socket;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ClientServerConnectionTest {
+    @Test
+    void readsInputStream() throws IOException {
+        String input = "Connected";
+        Socket socket = new FakeSocket(input);
+        ClientServerConnection clientServerConnection = new ClientServerConnection(socket);
+
+        assertEquals(input, clientServerConnection.read());
+    }
+
+    @Test
+    void WritesToOutputStream() throws IOException {
```

Comment:
I *think* the convention is to use camel case for JUnit tests. i.e. `writesToOutputStream`. 

My personal preference is to prefix my tests with "it` so that you end up with code that kind of reads like

_test it writes to the output stream_
```java
@Test void itWritesToTheOutputStream() {}
```

Reply:
Will make the change